packages/@lwc/engine-server/src/__tests__/fixtures/attribute-class/unstyled/array/expected.html

```diff
@@ -2,5 +2,9 @@
   <template shadowrootmode="open">
     <button class="button__icon">
     </button>
+    <x-child class="button__icon">
+      <template shadowrootmode="open">
+      </template>
+    </x-child>
```

Comment:
Another case we should cover here for safety: dynamic components (`lwc:component`). These technically go down a different code path (`LwcComponent.ts`), although you should still hit it since it will call into `getChildAttrsOrProps`.  As mentioned elsewhere though, I think this should be a separate standalone fixture.

Reply:
Seems like an unrelated error!

Reply:
Opened https://github.com/salesforce/lwc/pull/4958